### PR TITLE
Time Series Classification

### PR DESCRIPTION
**Pull request by brianleahy**
A univariate time-series classification task project.  Please review and suggest any changes.

### CODE REVIEW

File: tasks/time-series/time-series-classification/Time-Series-Classifier.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+# Acknowledgement for the dataset included in this package:
+@misc{UCRArchive2018,
+    title = {The UCR Time Series Classification Archive},
+    author = {Dau, Hoang Anh and Keogh, Eamonn and Kamgar, Kaveh and Yeh, Chin-Chia Michael and Zhu, Yan
+              and Gharghabi, Shaghayegh and Ratanamahatana, Chotirat Ann and Yanping and Hu, Bing
+              and Begum, Nurjahan and Bagnall, Anthony and Mueen, Abdullah and Batista, Gustavo},
+    year = {2018},
+    month = {October},
+    note = {\url{https://www.cs.ucr.edu/~eamonn/time_series_data_2018/}}
+}
```
Comment (slavakurilyak) [inline tasks/time-series/time-series-classification/Time-Series-Classifier.py]: Please turn lines into a comment with `#`

File: tasks/time-series/time-series-classification/Time-Series-Classifier.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            help="Optimizer, train only (%s)" % optimizer_list[0], type=str)
+    parser.add_argument('--epochs', default=300, help="Number of epochs to train, train only (300)", type=int)
+    parser.add_argument('--model', default='latest', help="Pre-trained model filename in 'models' directory, classify only (latest)", type= str)
+    return parser.parse_args()
+
+# Load dataset named base_name that is in the UCR Time Series Classification Archive format
+# see https://www.cs.ucr.edu/%7Eeamonn/time_series_data_2018/
+def load_UCR_dataset(directory, base_name):
+    train_filename = (directory + "\\" + base_name + "\\" + base_name + trainingset_ext)
+    test_filename = (directory + "\\" + base_name + "\\" + base_name + testset_ext)
```
Comment (slavakurilyak) [inline tasks/time-series/time-series-classification/Time-Series-Classifier.py]: For Linux/MacOS compatibility, please change this to:

```
train_filename = (directory + "/" + base_name + "/" + base_name + trainingset_ext)
test_filename = (directory + "/" + base_name + "/" + base_name + testset_ext)
```

File: tasks/time-series/time-series-classification/Time-Series-Classifier.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    # test_output = keras.layers.Conv1D(128, 3, activation='relu')(test_input)
+    # test_output = keras.layers.Conv1D(128, 3, activation='relu')(test_output)
+    # test_output = keras.layers.MaxPooling1D(3)(test_output)
+    # ...etc.
+    # test_model = keras.models.Model(inputs=test_input, outputs=test_output)
+
+if __name__ == '__main__':
+    # Clear old session
+    # TODO: Check that --help wasn't used on command line first?
+    K.clear_session()
```
Comment (slavakurilyak) [inline tasks/time-series/time-series-classification/Time-Series-Classifier.py]: @brianleahy this line fails if running on Theano

**Comment by slavakurilyak** [comment]
@brianleahy Thanks for making these modifications

**Comment by brianleahy** [comment]
@slavakurilyak  You're welcome, and thanks for pointing them out.  I changed the code to use the 'os' package feature to ensure the path separators will be correct for each operating system.  I checked OS X and Windows, but not Linux.  If you or someone else could verify Linux that would be great.

**Comment by slavakurilyak** [comment]
@mlaradji Please confirm this model works on Linux

**Comment by brianleahy** [comment]
Has this been reviewed yet?  GitHub just marked it stale for inactivity, and I see that there's still no other time-series classification contribution in the project.

**Pull request closed without merging**